provider/win_driver_installer_test.py

```diff
@@ -0,0 +1,143 @@
+                    '"ACPI\\QEMU0002"']
+
+device_name_list = ["Red Hat VirtIO Ethernet Adapter", "VirtIO RNG Device",
+                    "VirtIO Serial Driver", "VirtIO Balloon Driver",
+                    "QEMU PVPanic Device", "VirtIO Input Driver",
+                    "VirtIO FS Device", "Red Hat VirtIO SCSI controller",
+                    "Red Hat VirtIO SCSI pass-through controller",
+                    "QEMU FWCfg Device"]
+
+
+def install_gagent(session, test, qemu_ga_pkg, gagent_install_cmd):
+    """Install guest agent."""
+    logging.info("Install 'qemu-guest-agent' package in guest.")
+    vol_virtio_key = "VolumeName like '%virtio-win%'"
+    vol_virtio = utils_misc.get_win_disk_vol(session, vol_virtio_key)
+    qemu_ga_pkg_path = r"%s:\%s\%s" % (vol_virtio, "guest-agent", qemu_ga_pkg)
+    gagent_install_cmd = gagent_install_cmd % qemu_ga_pkg_path
+    s_inst, o_inst = session.cmd_status_output(gagent_install_cmd)
+    if s_inst != 0:
+        test.fail("qemu-guest-agent install failed,"
+                  " the detailed info:\n%s." % o_inst)
+
+
+def uninstall_gagent(session, test, gagent_uninstall_cmd):
+    """Uninstall guest agent."""
+    logging.info("Try to uninstall 'qemu-guest-agent' package.")
+    s, o = session.cmd_status_output(gagent_uninstall_cmd)
+    if s:
+        test.fail("Could not uninstall qemu-guest-agent package ")
+
+
+def get_gagent_version(session, test, qemu_ga_pkg, gagent_install_cmd,
+                       gagent_pkg_info_cmd):
+    """Get guest agent version."""
+    install_gagent(session, test, qemu_ga_pkg, gagent_install_cmd)
+    gagent_version = session.cmd_output(gagent_pkg_info_cmd).split()[-2]
+    return gagent_version
+
+
+def win_uninstall_all_drivers(session, test, params):
+    """Uninstall all drivers from windows guests"""
+    devcon_path = params["devcon_path"]
+    if params.get("check_qemufwcfg", "no") == "yes":
+        driver_name_list.append('qemufwcfg')
+    for driver_name, device_name, device_hwid in zip(driver_name_list,
+                                                     device_name_list,
+                                                     device_hwid_list):
+        win_driver_utils.uninstall_driver(session, test, devcon_path,
+                                          driver_name, device_name,
+                                          device_hwid)
+
+
+def win_installer_test(session, test, params, vm):
+    """windows installer related check"""
+    run_install_cmd = params["run_install_cmd"]
+    installer_pkg_check_cmd = params["installer_pkg_check_cmd"]
+    error_context.context("Install virtio-win drivers via "
+                          "virtio-win-guest-tools.exe.", logging.info)
+    vm.send_key('meta_l-d')
+    time.sleep(30)
+    win_driver_utils.install_driver_by_installer(session, test,
+                                                 run_install_cmd,
+                                                 installer_pkg_check_cmd)
```

Comment:
This seems could be abstract into a function, with an extra parameter `copy_files_params=None`, something like:
```suggestion
def _install_test(vm, session, test, run_install_cmd, installer_pkg_check_cmd, copy_files_params=None):
    vm.send_key('meta_l-d')
    time.sleep(30)
    if copy_files_params:
        win_driver_utils.copy_file_to_samepath(session, test, copy_files_params)
    win_driver_utils.install_driver_by_installer(session, test,
                                                 run_install_cmd,
                                                 installer_pkg_check_cmd)
```

Then you might reference it in the other scripts.